src/pages/registration/contributors_tab/Authors.tsx

```diff
@@ -0,0 +1,130 @@
+import { FieldArrayRenderProps, FormikProps, useFormikContext, move } from 'formik';
+import React, { FC, useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setNotification } from '../../../redux/actions/notificationActions';
+import { Authority } from '../../../types/authority.types';
+import { Contributor, emptyContributor, UnverifiedContributor } from '../../../types/contributor.types';
+import { NotificationVariant } from '../../../types/notification.types';
+import { BackendTypeNames } from '../../../types/publication_types/commonRegistration.types';
+import { Registration } from '../../../types/registration.types';
+import { overwriteArrayMerge } from '../../../utils/formik-helpers';
+import AuthorTable from './components/AuthorTable';
+import AddIcon from '@material-ui/icons/Add';
+import styled from 'styled-components';
+import { Button } from '@material-ui/core';
+import AddContributorModal from './AddContributorModal';
+
+const StyledAddAuthorButton = styled(Button)`
+  margin-top: 1rem;
+`;
+
+interface AuthorsProps extends Pick<FieldArrayRenderProps, 'push' | 'replace'> {}
+
+const Authors: FC<AuthorsProps> = ({ push, replace }) => {
+  const { t } = useTranslation('registration');
+  const dispatch = useDispatch();
+  const { values, setValues }: FormikProps<Registration> = useFormikContext();
+  const {
+    entityDescription: { contributors },
+  } = values;
+  const [openContributorModal, setOpenContributorModal] = useState(false);
+  const [unverifiedAuthor, setUnverifiedAuthor] = useState<UnverifiedContributor | null>(null);
+  const orderedAuthors = [...contributors].map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+
+  const toggleContributorModal = () => {
+    if (unverifiedAuthor) {
+      setUnverifiedAuthor(null);
+    }
+    setOpenContributorModal(!openContributorModal);
+  };
+
+  useEffect(() => {
+    if (unverifiedAuthor) {
+      // Open modal if user has selected an unverified author
+      setOpenContributorModal(true);
+    }
+  }, [unverifiedAuthor]);
+
+  const handleOnRemove = (indexToRemove: number) => {
+    const remainingAuthors = [...contributors]
+      .filter((_, index) => index !== indexToRemove)
+      .map((contributor, index) => ({ ...contributor, sequence: index + 1 }));
+    setValues(
+      deepmerge(values, { entityDescription: { contributors: remainingAuthors } }, { arrayMerge: overwriteArrayMerge })
+    );
+  };
+
+  const handleMoveAuthor = (newIndex: number, oldIndex: number) => {
+    const reorderedAuthors = move(orderedAuthors, oldIndex, newIndex) as Contributor[];
+    // Ensure incrementing sequence values
+    const newAuthors = reorderedAuthors.map((contributor, index) => ({
+      ...contributor,
+      sequence: index + 1,
+    }));
+    setValues(
+      deepmerge(values, { entityDescription: { contributors: newAuthors } }, { arrayMerge: overwriteArrayMerge })
```

Comment:
FieldArray har jo metoden `move` for dette, men mulig vi har gjort det slik siden vi uansett må oppdatere `sequence` for alle

Reply:
alle default-metoder fra FieldArray har sikkert vært kjempefine å bruke hvis vi ikke hadde brukt `sequence` til noe